crates/oxc_semantic/src/node.rs

```diff
@@ -151,8 +151,9 @@ impl<'s, 'a> Iterator for AstNodeParentIter<'s, 'a> {
     type Item = &'s AstNode<'a>;
 
     fn next(&mut self) -> Option<Self::Item> {
+        let next = self.curr;
```

Comment:
why do we call it `self.curr` if it signifies next?

Reply:
Yeah I wasn't sure exactly how to name it, since it's produced by next but is the "current" parent while you're iterating over it